### PR TITLE
partition: fix unstable test TestExchangePartitionStates

### LINKED ISSUE #46815: unstable test TestExchangePartitionStates
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Run unit test.

```
go test --tags=intest --run="TestExchangePartitionStates"
```

### 2. What did you expect to see? (Required)

no error.

### 3. What did you see instead (Required)

````
                Test:       	TestExchangePartitionStates
        	Messages:   	sql:insert into t values (1000006,"1000006"), [], error stack [table:1748]Found a row not matching the given partition set
        	            	github.com/pingcap/tidb/table/tables.(*partitionedTable).CheckForExchangePartition
        	            		table/tables/partition.go:1278
        	            	github.com/pingcap/tidb/executor.checkRowForExchangePartition
        	            		executor/write.go:329
        	            	github.com/pingcap/tidb/executor.(*InsertValues).fillRow
        	            		executor/insert_common.go:696
        	            	github.com/pingcap/tidb/executor.(*InsertValues).fastEvalRow
        	            		executor/insert_common.go:411
        	            	github.com/pingcap/tidb/executor.insertRows
        	            		executor/insert_common.go:219
        	            	github.com/pingcap/tidb/executor.(*InsertExec).Next
        	            		executor/insert.go:307
        	            	github.com/pingcap/tidb/executor/internal/exec.Next
        	            		executor/internal/exec/executor.go:283
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).next
        	            		executor/adapter.go:1223
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelayExecutor
        	            		executor/adapter.go:968
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).handlePessimisticDML
        	            		executor/adapter.go:1029
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).handleNoDelay
        	            		executor/adapter.go:791
        	            	github.com/pingcap/tidb/executor.(*ExecStmt).Exec
        	            		executor/adapter.go:575
        	            	github.com/pingcap/tidb/session.runStmt
        	            		session/session.go:2420
        	            	github.com/pingcap/tidb/session.(*session).ExecuteStmt
        	            		session/session.go:2270
        	            	github.com/pingcap/tidb/testkit.(*TestKit).ExecWithContext
        	            		testkit/testkit.go:342
        	            	github.com/pingcap/tidb/testkit.(*TestKit).MustExecWithContext
        	            		testkit/testkit.go:134
        	            	github.com/pingcap/tidb/testkit.(*TestKit).MustExec
        	            		testkit/testkit.go:129
        	            	table/tables/test/partition/partition_test.TestExchangePartitionStates
        	            		table/tables/test/partition/partition_test.go:770
        	            	testing.tRunner
        	            		GOROOT/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		src/runtime/asm_amd64.s:1650

```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

master

### PR DESCRIPTION
**Pull request by jiyfhust**
### What problem does this PR solve?

Issue Number: close #46815

Problem Summary:

### What is changed and how it works?

Two ways to reproduce the error:
1. run many times, let the competitive occur:
```
$ go test --tags=intest --run="TestExchangePartitionStates" 
```
2.  make the competitive easier to occur: 
```
$ git diff
diff --git a/table/tables/test/partition/partition_test.go b/table/tables/test/partition/partition_test.go
index 2f218db..702a0d1 100644
--- a/table/tables/test/partition/partition_test.go
+++ b/table/tables/test/partition/partition_test.go
@@ -749,7 +749,7 @@ func TestExchangePartitionStates(t *testing.T) {
                                logutil.BgLogger().Info("Got state", zap.String("State", s))
                                break
                        }
-                       gotime.Sleep(50 * gotime.Millisecond)
+                       gotime.Sleep(1 * gotime.Millisecond)
                }
        }
        waitFor("t", "write only", 4)

$ go test --tags=intest --run="TestExchangePartitionStates" 

```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: table/tables/test/partition/partition_test.go [hunk 1]
```diff
@@ -747,6 +747,8 @@ func TestExchangePartitionStates(t *testing.T) {
 			res := tk4.MustQuery(`admin show ddl jobs where db_name = '` + strings.ToLower(dbName) + `' and table_name = '` + tableName + `' and job_type = 'exchange partition'`).Rows()
 			if len(res) == 1 && res[0][pos] == s {
 				logutil.BgLogger().Info("Got state", zap.String("State", s))
+				// Sleep 50ms to wait load InforSchema finish, issue #46815.
```
Comment (zimulala) [inline table/tables/test/partition/partition_test.go:750]: Maybe adding a retry is better？

**Review by mjonss — APPROVED** [review]
Seems like an improvement. Would it be possible to actually check and wait for the infoSchema update?

**Comment by jiyfhust** [comment]
> Seems like an improvement. Would it be possible to actually check and wait for the infoSchema update?

I did not find a method to check it properly.

**Comment by jiyfhust** [comment]
PTAL @zimulala

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**